Expand resolver documentation (#3802)
ChilliCream/hotchocolate: website/src/docs/hotchocolate/fetching-data/index.md

---
title: Overview
---

In this section we will learn everything about fetching data with Hot Chocolate.

# Resolvers

> Resolvers are the main building blocks when it comes to fetching data. Every field in our GraphQL schema is backed by such a resolver function, responsible for fetching the data associated with this field. Since a resolver is just a function we can use it to retrieve data from a database, a REST service or any other data source.

Reviewer: ```suggestion
Resolvers are the main building blocks when it comes to fetching data. Every field in our GraphQL schema is backed by such a resolver function, responsible for returning the field's value. Since a resolver is just a function, we can use it to retrieve data from a database, a REST service, or any other data source as needed. 
```

[Learn more about DataLoaders](/docs/hotchocolate/fetching-data/dataloader)

# Pagination

Hot Chocolate provides pagination capabilities out of the box. They allow us to expose pagination in a standardized way and can even take care of crafting the necessary pagination queries to our databases.

[Learn more about pagination](/docs/hotchocolate/fetching-data/pagination)

# Filtering

When returning a list of entites, we often need to filter them using operations like `equals`, `contains`, `startsWith`, etc. Hot Chocolate takes away a lot of the boilerplate, by handling the generation of necessary input types and even translating the applied filters into native database queries.

[Learn more about filtering](/docs/hotchocolate/fetching-data/filtering)

# Sorting

Similar to filtering, Hot Chocolate can also autogenerate input types related to sorting. They allow us to specify by which fields and in which direction our entities should be sorted. These can also be translated into native database queries automatically.

[Learn more about sorting](/docs/hotchocolate/fetching-data/sorting)

# Projections

> Projections allow Hot Chocolate to project incoming GraphQL queries to the database.

Reviewer: ```suggestion
Projections allow Hot Chocolate to project incoming GraphQL queries to optimal database queries.
```

Reviewer: Optimal might be promising too much, it's really just optimal in the sense of the cols selected...

> If we for example only select the `Name` and `Id` of a user in our GraphQL query, Hot Chocolate will only select those two fields when querying the user in the database.

Reviewer: ```suggestion
For example, if the client only requests the `name` and `id` of a user in their GraphQL query, Hot Chocolate will only query the database for those two columns.
```

Do you also want to mention it can handle relational properties?
